pkg/controller/pipeline/reconciler.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 VMware
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/cartographer/pkg/conditions"
+	realizerpipeline "github.com/vmware-tanzu/cartographer/pkg/realizer/pipeline"
+	"github.com/vmware-tanzu/cartographer/pkg/repository"
+)
+
+type Reconciler interface {
+	Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error)
+}
+
+func NewReconciler(repository repository.Repository, realizer realizerpipeline.Realizer) Reconciler {
+	return &reconciler{
+		repository: repository,
+		realizer:   realizer,
+	}
+}
+
+type reconciler struct {
+	repository repository.Repository
+	realizer   realizerpipeline.Realizer
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := logr.FromContext(ctx).
+		WithValues("name", request.Name, "namespace", request.Namespace)
+	logger.Info("started")
+
+	pipeline, getPipelineError := r.repository.GetPipeline(request.Name, request.Namespace)
```

Comment:
any reason for not using `err` here?

usually errors are not named as `err` when there's a very good reason for it (e.g., to not replace an error that was captured before that we want to return at a later point ...), but this doesn't seem to be the case

```suggestion
	pipeline, err := r.repository.GetPipeline(request.Name, request.Namespace)
```